core/src/toga/style/applicator.py

```diff
@@ -25,7 +25,16 @@ def set_text_alignment(self, alignment):
     def set_hidden(self, hidden):
         self.widget._impl.set_hidden(hidden)
         for child in self.widget.children:
-            child.applicator.set_hidden(hidden)
+            # If the parent is hidden, then so are all children. However, if the
+            # parent is visible, then the child's explicit visibility style is
+            # taken into account. This visibility cascades into any
+            # grandchildren.
+            #
+            # parent hidden child hidden style child final hidden state
+            # ============= ================== ======================== True
+            #  True               True True          False              True
+            #  False         True               True False         False False
```

Comment:
```suggestion
            # parent hidden child hidden style child final hidden state
            # ============= ================== ======================== 
            # True          True               True
            # True          False              True
            # False         True               True 
            # False         False              False
```

Reply:
... How did *that* happen? I swear these were all lined up when I committed...